packages/teams-js/src/private/files.ts

```diff
@@ -710,40 +731,90 @@ export namespace files {
    * @hidden
    * Hide from docs
    *
-   * Initiates the 3P cloud storage provider file action (Upload / Download / Delete) flow
+   * Initiates the delete 3P cloud storage file(s) flow, which will delete existing file(s) from the given 3P provider
+   *
+   * @param deleteFileRequest 3P cloud storage provider delete action request content
+   * @param callback Callback that will be triggered post deleting existing file(s) flow is finished
+   */
+  export function deleteCloudStorageProviderFile(
+    deleteFileRequest: CloudStorageProviderRequest<CloudStorageProviderDeleteFileContent>,
+    callback: (error?: SdkError) => void,
+  ): void {
+    ensureInitialized(FrameContexts.content);
+
+    if (!callback) {
+      throw getSdkError(ErrorCode.INVALID_ARGUMENTS, '[files.deleteCloudStorageProviderFile] callback cannot be null');
+    }
+
+    if (!(deleteFileRequest && deleteFileRequest.content)) {
```

Comment:
shouldn't you also be checking that itemList is present / non-null/non-empty/all CloudStorageFolderItems are files and not directories